### PR TITLE
Settings for which things to sync

### LINKED ISSUE #54: Settings for which things to sync
Add ability to check/uncheck "categories" for syncing (i.e. "theme", "packages", "package settings", etc.).

### PR DESCRIPTION
**Pull request by jerone**
* [x] Closes #54
* [x] Settings are ordered
* [x] Updated specs
* [x] Write new specs

Not happy with the `cb` logic, but that's another issue

sync-settings

### CODE REVIEW

**Comment by jerone** [comment]
Excuses for the labels and reopening, was trying something with [waffle.io](https://waffle.io/Hackafe/atom-sync-settings).

File: lib/sync-settings.coffee [hunk 1]
```diff
@@ -36,19 +63,19 @@ module.exports =
...
+    if atom.config.get('sync-settings.syncPackages')
+      files["packages.json"] = content: JSON.stringify(@getPackages(), null, '\t')
+    if atom.config.get('sync-settings.syncKeymap')
+      files["keymap.cson"] = content: (@fileContent atom.keymaps.getUserKeymapPath()) ? "# keymap file (not found)"
+    if atom.config.get('sync-settings.syncStyles')
+      files["styles.less"] = content: (@fileContent atom.styles.getUserStyleSheetPath()) ? "// styles file (not found)"
+    if atom.config.get('sync-settings.syncInit')
+      files["init.coffee"] = content: (@fileContent atom.config.configDirPath + "/init.coffee") ? "# initialization file (not found)"
+    if atom.config.get('sync-settings.syncSnippets')
+      files["snippets.cson"] = content: (@fileContent atom.config.configDirPath + "/snippets.cson") ? "# snippets file (not found)"
```
Comment (groupsky) [inline lib/sync-settings.coffee]: Can we replace with a map like:

```coffeescript
fileConfig = 
  'settings.json': => JSON.stringify(atom.config.settings, @filterSettings, '\t')
```

and then just iterate over it like
```coffeescript
files = for file, reader of fileConfig
  reader() if atom.config.get("sync-settings.sync#{_.capitalize file}")
```


The map could be extended to include also a writer, and then it can be used in both backup and restore.

Comment (jerone) [inline lib/sync-settings.coffee]: I actually have plans to separate packages into core packages, [community packages](https://github.com/Hackafe/atom-sync-settings/issues/37) and themes.

Comment (groupsky) [inline lib/sync-settings.coffee]: That sounds better for user configuration, and still the code can be refactored. But the refactoring should be in a different PR. No need to blur this one.

Comment (jerone) [inline lib/sync-settings.coffee]: @groupsky Maybe I miss interpreted your note above, did you mean you wanted get ride of the `content` property or make all settings loopable?

Comment (groupsky) [inline lib/sync-settings.coffee]: @jerone the latter. Having a lot of `if`s and duplicated code doesn't look nice and should be refactored in a more elegant form. That was my suggestion. But the refactoring should not take place in this PR. I'll open a new issue to track that.

**Comment by groupsky** [comment]
I think the setting should be honored also when restoring.

**Comment by groupsky** [comment]
:+1:

File: lib/sync-settings.coffee (hunk 1 above)
Comment (jerone) [inline lib/sync-settings.coffee]: https://github.com/Hackafe/atom-sync-settings/issues/79 :+1:

**Comment by timfallmk** [comment]
:+1:
